metashare/api/models.py

```diff
@@ -55,18 +55,32 @@ class User(HashIdMixin, AbstractUser):
     currently_fetching_repos = models.BooleanField(default=False)
     devhub_username = StringField(null=True, blank=True)
 
+    def queue_refresh_repositories(self):
+        """Queue a job to refresh repositories unless we're already doing so"""
+        from .jobs import refresh_github_repositories_for_user_job
+
+        if not self.currently_fetching_repos:
+            self.currently_fetching_repos = True
+            self.save()
+            refresh_github_repositories_for_user_job.delay(self)
+
     def refresh_repositories(self):
-        repos = gh.get_all_org_repos(self)
-        GitHubRepository.objects.filter(user=self).delete()
-        GitHubRepository.objects.bulk_create(
-            [
-                GitHubRepository(user=self, repo_id=repo.id, repo_url=repo.html_url)
-                for repo in repos
-            ]
-        )
-        self.currently_fetching_repos = False
-        self.save()
-        self.notify_repositories_updated()
+        try:
+            repos = gh.get_all_org_repos(self)
+            with transaction.atomic():
+                GitHubRepository.objects.filter(user=self).delete()
+                GitHubRepository.objects.bulk_create(
+                    [
+                        GitHubRepository(
+                            user=self, repo_id=repo.id, repo_url=repo.html_url
+                        )
+                        for repo in repos
+                    ]
+                )
+            self.notify_repositories_updated()
```

Comment:
In this case, I think we actually want this in the `finally` block. What this does on the front end is 1) set `user.currently_fetching_repos = false` and 2) refetch the list of repos from the API. I think even in the case of an error (1) is desired, and (2) doesn't hurt.

Reply:
Addressed in https://github.com/oddbird/MetaShare/commit/57984c9c02b0bfacd50d375374f7f05693558b91